File: app/src/main/java/com/frolo/muse/engine/SimpleAudioFxObserver.java
package com.frolo.muse.engine;

import com.frolo.muse.model.preset.Preset;
import com.frolo.muse.model.reverb.Reverb;


/**
 * Simple implementation of {@link AudioFxObserver} that does nothing in callback methods.
 */
public abstract class SimpleAudioFxObserver implements AudioFxObserver {
    @Override
    public void onEnabled(AudioFx audioFx) {
    }

    @Override
    public void onDisabled(AudioFx audioFx) {
    }

    @Override
    public void onBandLevelChanged(AudioFx audioFx, short band, short level) {
    }

    @Override
    public void onPresetUsed(AudioFx audioFx, Preset preset) {
    }

    @Override
    public void onBassStrengthChanged(AudioFx audioFx, short strength) {
    }

    @Override
    public void onVirtualizerStrengthChanged(AudioFx audioFx, short strength) {
    }

    @Override
    public void onReverbUsed(AudioFx audioFx, Reverb reverb) {
    }
}

File: app/src/main/java/com/frolo/muse/engine/SimpleAudioFxObserver.kt
package com.frolo.muse.engine

import com.frolo.muse.model.preset.Preset
import com.frolo.muse.model.reverb.Reverb


abstract class SimpleAudioFxObserver: AudioFxObserver {
    override fun onEnabled(audioFx: AudioFx) = Unit

    override fun onDisabled(audioFx: AudioFx) = Unit

    override fun onBandLevelChanged(audioFx: AudioFx, band: Short, level: Short) = Unit

    override fun onPresetUsed(audioFx: AudioFx, preset: Preset) = Unit

    override fun onBassStrengthChanged(audioFx: AudioFx, strength: Short) = Unit

    override fun onVirtualizerStrengthChanged(audioFx: AudioFx, strength: Short) = Unit

    override fun onReverbUsed(audioFx: AudioFx, reverb: Reverb) = Unit
}